equinox/_module.py

```diff
@@ -478,11 +480,44 @@ def __new__(
                                     "attempting to override "
                                     f"`{base.__module__}.{base.__qualname__}.{k}`."
                                 )
+
         # [Step 6] Register as a pytree.
+        data_fs = tuple(
+            f.name
+            for f in dataclasses.fields(cls)
+            if not f.metadata.get("static", False)
+        )
+        static_fs = tuple(
+            f.name for f in dataclasses.fields(cls) if f.metadata.get("static", False)
+        )
+
+        def flatten_module(
+            obj: "Module", /, with_keys: bool
+        ) -> tuple[tuple[PyTree, ...], _FlattenedData]:
+            # Static metadata, placed in aux.
+            static_vs = [getattr(obj, name) for name in static_fs]
```

Comment:
I think static fields could also fail to be initialised at times, e.g. when partway through `__init__`.

Also, I think this needs to be `obj.__dict__.get(name)` to avoid looking up through the class and the class hierarchy, in case of shadowing. We only want to grab instance attributes.